api/src/controllers/settings.ts

```diff
@@ -23,6 +23,32 @@ router.get(
 	respond
 );
 
+router.patch(
+	'/themes',
```

Comment:
@rijkvanzanten a bit off-topic here, but would there be merit on having such endpoint for the translations strings as well? Since it's currently one big json field when there's a lot, and may not be as straightforward if a user wishes to add/update a particular translation via API. Most of the heavy lifting was done App side. I did consider this before but opted out of it in that PR for simplicity sake.

Reply:
I see what you mean. System consistency is absolutely a good argument.

I still like a separate theme endpoint, but you're the boss 😜 Shouldn't be much of a hassle to get it switched up. I'll also remove the `update_theme_overrides` GraphQL endpoint that I added, in favor of the default settings endpoint.

On a more serious note, the backend validation of the theme structure is pretty important. Of course, if the structure is wrong, it should just fall back to the base theme, however it would also generate a bunch of unused variables on the page. Overall, I believe maintaining those checks is necessary. 

---

On a side note, I'd like to quickly explain a quirk with how the theme updates work. 

As it stands, in order to avoid modifying/deleting themes that haven't been edited (since it's all one big JSON object), only the theme roots that are passed to the api are handled. For instance, if the contents of `theme_overrides` is like:
```js
{
  "dark": {
    "global": { . . . },
    "components": { . . . },
  },
  "light": {
    "global": { . . . },
    "components": { . . . },
  },
}
```

Then, after editing the dark theme, the following is passed to the patch endpoint:
```js
{
  "dark": {
    "global": { . . . },
  },
}
```

The top-level theme names/types are effectively modified on an opt-in basis. So in this case, the `"light"` theme property will not be edited at all. However, the `"dark"` theme will be entirely overwritten with the contents therein. 

If an empty object is passed, it will not edit anything. However, if the following is passed, it will remove all dark theme overrides:
```js
{
  "dark": {},
}
```

--- 

If you have any questions or objections to this, I'm happy to discuss. In short, handling it this way makes the front-end logic simpler. We only have to pass the edits for the theme(s) that we're working on.

Reply:
I love the custom in-json-blob validation, though I don't think it should be a one-off for just a single field in settings. If we really want to utilize this "structured JSON blob with custom validation" functionality, we should implement it in a way where any collection/field in the system can utilize it, and then use it here in the system configuration. It's the same consistency principle 🙂 I'm basically trying real hard to make sure the system collections could've been all made through `/items/`, with minimal custom logic. As of right now, I'd prefer relying on an unstructured blob w/ defensive programming in the app rather than adding in a custom validator for just this one particular field. If the nested-json-validation is mission critical for this PR, we should figure out a standardized implementation of the validation piece first 🙂

Reply:
You make good points, and I agree. Typically my coding style is similar, I try to leverage a consistent foundation, modeling as much as possible in data that can be procedurally parsed, as opposed to hardcoding (I hope that description made sense 🤔 ). Suffice to say, if we could somehow add procedural JSON validation directly into the `items` service, I think that would elegantly solve this problem, as well as set us up to handle similar validations in the future.

As far as the criticality of validation in this instance, that's difficult to determine. Because the theming system, and subsequent variable generation, are so reliant on a specific structure to generate the correct CSS variable names, I currently have a few checks in place, both on the API side and front-end, in an attempt to cover all bases. The front-end checks should be enough, however, they of course don't account for custom API calls. On the flip side, as I mentioned before, a malformed JSON object doesn't break anything, it just doesn't get recognized by the theming interface, and it will generate unused CSS variable names. With that all said, I suppose it depends on how you define critical.

At the end of the day, building a JSON validator into the `items` service would only serve to improve the app, so I think it's worth doing. 

If we go that route, I think it would be worth while to look into other JSON validators. One I'm familiar with it called [AJV](https://ajv.js.org/). A huge benefit of AJV is that it utilizes standardized JSON schema formats, such as [JSON Schema](https://json-schema.org/). As a result, it's much less verbose than JOI. As well, it has a very nice interoperability with Typescript.

Reply:
I think for this particular PR (to keep the MVP state a little within scope), lets drop the custom validation logic in favor of figuring out a better "core" experience of configuring and enabling nested JSON validation 🙂 

As for AJV vs Joi, it's a tad subjective. When you're interfacing with AJV directly, it's a tad nicer to just give it a string representation of what you're trying to validate against, however Joi makes it a tad easier to programmatically "query build" together a schema object to then use in validation. As a matter of fact, we already have a pretty sturdy converter util that will generate a Joi schema for an input Directus Filter structure (eg `{ "name": { "_eq": "Rijk" }}` (which is actually optimized as we speak in https://github.com/directus/directus/pull/13596)

Reply:
For sure, that's definitely one area where JOI has the leg up, putting it together is a bit more human-readable. 

To be honest, I almost didn't even mention AJV because of how integral JOI is to the filters. It would definitely be a hard sell 😂 . But I figured it'd be better to put it out there and not need it. Definitely something to keep in mind for the future, never know when it could be useful.

Reply:
Oh for sure, I looked into it extensively when putting together the first iterations of the API 🙂